js/index.js

```diff
@@ -1 +1,120 @@
 // Your code goes here
+// prevent default
+
+const atags = document.getElementsByClassName('nav');
+console.log(atags)
+for (var i = 0 ; i < atags.length; i++) {
+    atags[i].addEventListener('click' , function(event){
+        console.log('link was clicked')
+        event.preventDefault();
+        alert('Error!')
+    }); 
+}
+
+
+
+//mouseover
+const buspic = document.querySelector('img');
+console.log(buspic);
+buspic.addEventListener('mouseover', function(event){
+    console.log('Mouse over');
+    buspic.style.visibility = 'hidden';
+});
+
+//mouseleave
+const head = document.getElementsByClassName('intro')
+console.log(head[0])
+head[0].addEventListener('mouseleave', function(event){
+    console.log('Mouse left')
+    buspic.style.visibility = 'visible';
+});
+
+
+//keydown 
+const Wpage = document.querySelector('body');
+Wpage.addEventListener('keydown', event => {
+    console.log('This key was pressed:', event.key)  
+})
+
+
+//load
+window.addEventListener('load', (event) => {
+    console.log('page is fully loaded');
+});
+
+//keyup
+ Wpage.addEventListener('keyup', event => {
+    if (event.key === 'ArrowRight'){
+        if (Wpage.style.backgroundColor ==='red'){
+            Wpage.style.backgroundColor = 'white';
+        }
+        else {
+            Wpage.style.backgroundColor = 'red';
+        }
+    }  
+  })
+
+//click 
+const footertxt = document.querySelector('footer p')
+console.log(footertxt)
+
+footertxt.addEventListener('click', event =>{
+    console.log('Footer text was clicked')
+    footertxt.style.visibility = 'hidden';
+})
+
+//dblclick 
+const footer = document.querySelector('footer')
+footer.addEventListener('dblclick', event => {
+    console.log('footer was clicked x2')
+    footertxt.style.visibility = 'visible'
+})
+
+//close 
+window.onbeforeunload = confirmExit;
+function confirmExit() {
+    return " "
+}
+
+// window.addEventListener('beforeunload', function (event) {
+//     event.preventDefault();
+//     event.returnValue = true;
+//   });
+
+// window.onbeforeunload = function(){
+//     return 'Yes?'
+// }
+
+//wheel
+Wpage.addEventListener('wheel' , function(event){
+    console.log('scrolling')
+    Wpage.style.backgroundColor = 'green' 
+}); ///is there anyway to check if the event is active 
+///i was trying to only have this work while the user is scrollling but i couldnt figure it out
+///somthing like if (event === true )
+
```

Comment:
This is a tricky effect to implement - you can't actually check to see if a `wheel` event is 'active' - a `wheel` event simply fires once every time you 'step'the mouse wheel forward or backward by a certain degree (some mouse models actually make it easier for you to feel each 'step' when you move the mouse wheel).  The `wheel` event is not a continuous event; it is simply just the same event repeatedly firing several times in quick succession.

What I would suggest is making using of both [`setTimeout`](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout) and [`clearTimeout`](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/clearTimeout) to create the illusion of making the `wheel` event continuous - after every iteration of the `wheel` event, reset the background color to be white within 1/4 of a second, but also reset this 1/4 second timeout if the mouse wheel is still moving. Sample implementation below:
```
let reset = 0;
Wpage.addEventListener('wheel' , function(event){
    console.log('scrolling')
    Wpage.style.backgroundColor = 'green' 
    clearTimeout(reset);
    reset = setTimeout(() => Wpage.style.backgroundColor = "white", 250);
});
```